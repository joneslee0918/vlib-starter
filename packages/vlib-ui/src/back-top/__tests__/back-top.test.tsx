import { nextTick } from 'vue'
import { mount } from '@vue/test-utils'
import { describe, test, expect } from 'vitest'
import { BackTop } from '../index'

describe('BackTop.vue', () => {
  test('render', async () => {
    const wrapper = mount(
      () => (
        <div class="target" style="height: 100px; overflow: auto">
          <div style="height: 10000px; width: 100%">
            <BackTop
              target=".target"
              visibilityHeight={2000}
              right={100}
              bottom={200}
            />
          </div>
        </div>
      ),
      { attachTo: document.body }
    )
    await nextTick()

    expect(wrapper.find('.vlib-back-top').exists()).toBe(false)

    wrapper.element.scrollTop = 2000
    await wrapper.trigger('scroll')
    expect(wrapper.find('.vlib-back-top').exists()).toBe(true)

    expect(wrapper.find('.vlib-back-top').attributes('style')).toBe(
      'right: 100px; bottom: 200px;'
    )

    await wrapper.trigger('click')
    expect(wrapper.emitted('click')).toBeDefined()

    expect(wrapper.html()).toMatchSnapshot()
  })
})
