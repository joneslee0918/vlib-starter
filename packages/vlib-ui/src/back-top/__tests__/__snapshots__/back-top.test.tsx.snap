// Vitest Snapshot v1

exports[`BackTop.vue > render 1`] = `
"<div class=\\"target\\" style=\\"height: 100px; overflow: auto;\\">
  <div style=\\"height: 10000px; width: 100%;\\">
    <transition-stub>
      <div class=\\"vlib-back-top\\" style=\\"right: 100px; bottom: 200px;\\"><button type=\\"button\\" class=\\"van-button van-button--primary van-button--normal van-button--round\\" disabled=\\"false\\" vlib-back-top=\\"\\">
          <div class=\\"van-button__content\\"><i class=\\"van-badge__wrapper van-icon van-icon-back-top van-button__icon\\">
              <!---->
              <!---->
              <!---->
            </i>
            <!---->
            <!---->
          </div>
        </button></div>
    </transition-stub>
  </div>
</div>"
`;
